lib/puppet/module/task.rb

```diff
@@ -0,0 +1,85 @@
+require 'json'
+
+class Puppet::Module
+  class Task
+    class Error < Puppet::Error; end
+    class InvalidName < Error; end
+    class InvalidFile < Error; end
+    class TaskNotFound < Error; end
+
+    FORBIDDEN_EXTENSIONS = %w{.conf .md}
+
+    def self.is_task_name?(name)
+      return true if name =~ /^[a-z][a-z0-9_]*$/
+      return false
+    end
+
+    # Determine whether a file has a legal name for either a task's executable or metadata file.
+    def self.is_tasks_filename?(path)
+      name_less_extension = File.basename(path, '.*')
```

Comment:
Can you replace this (and others) with `Puppet::FileSystem.basename`? We generally have augmented/re-implemented some ruby filesystem manipulation (for better or worse) to allow us to improve platform-specific support. `basename` isn't particularly controversial but it's generally asked for to use the Puppet::FileSystem equivalents if they exist.